src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ApplicationSettingsBase.cs

```diff
@@ -725,6 +725,9 @@ private object GetPropertyValue(string propertyName)
                     }
                 }
 
+                // we query the value first so that we initialize all values from value providers and so that we don't end up
+                // on an infinite recursion when calling Properties[propertyName] as that calls this.
+                object _ = base[propertyName];
```

Comment:
(nit) can you just use 

```suggestion
                _ = base[propertyName];
```